code/game/objects/effects/decals/cleanable/aliens.dm

```diff
@@ -39,9 +39,12 @@
 	if(!step_to(src, get_step(src, direction), 0))
 		return
 	if(mapload)
-		for (var/i in 1 to range)
+		for (var/i = 1, i < range, i++)
+			var/turf/turf_sending = get_step(src, direction)
+			if(isclosedturf(turf_sending) || (isgroundlessturf(turf_sending) && !SSmapping.get_turf_below(turf_sending)))
+				continue
 			new /obj/effect/decal/cleanable/xenoblood/xsplatter(loc)
-			if (!step_to(src, get_step(src, direction), 0))
+			if (!step_to(src, turf_sending, 0))
```

Comment:
bit weird to have closed turf here, aren't we ok with bumping up against a closed turf? s why we paint as we leave a tile and all

Reply:
isn't the point of using step_to to avoid allowing them to enter a turf?

Reply:
You're right, and my previous comment is stupid. It was just that I copy pasted this from every other instance of checking for groundless turf and such. I put it in to prevent blood from streaking into space, and just used the same check I used elsewhere.

Reply:
Ok.
Might want to move the check a line down, so we can properly lay gore on the tile BEFORE an "unacceptable" turf (wall, etc)
also uh, should drop the wall check since we want to break in that case I think

Reply:
You mean like this?
![image](https://github.com/tgstation/tgstation/assets/53777086/4ff26553-6c96-4a2e-892c-c42d288467d3)

Reply:
yes